swig/customCode/NativeLibLoader.java

```diff
@@ -323,13 +324,12 @@ private static Path findNativeLibrary(String libraryName, boolean mapLibraryName
     String mappedLibraryName = mapLibraryName ? System.mapLibraryName(libraryName) : libraryName;
     String libDir = LIB_RESOURCE_DIR;
 
-    if (System.getProperty("os.name").contains("mac")
+    if (System.getProperty("os.name").contains("Mac")
         && System.getProperty("os.arch").equals("aarch64")) {
-      libDir = "arm" + libDir;
+      libDir = "/arm" + libDir;
```

Comment:
Because our tests were wrongly using and x86_64 version of Java. This was causing to run under Rosetta and as a consequence the path to load the arm native libs was never used/tested. So overall even though we were shipping the our release with the arm native libs correctly the NativeLibLoader was not using them.